src/Microsoft.Azure.WebJobs/StorageAccountAttribute.cs

```diff
@@ -27,14 +27,22 @@ public sealed class StorageAccountAttribute : Attribute
         ///   to any INameResolver registered on the JobHostConfiguration to resolve the actual setting name to use.
         /// - Can be an app setting or connection string name of your choosing.
         /// </param>
-        public StorageAccountAttribute(string account)
+        /// <param name="supported">A flags enum indicating which types of Azure Storage accounts are
+        /// supported by this attribute.</param>
+        public StorageAccountAttribute(string account, StorageAccountTypes supported = StorageAccountTypes.All)
         {
             Account = account;
+            Supported = supported;
         }
 
         /// <summary>
         /// Gets the name of the Azure Storage connection string to use.
         /// </summary>
         public string Account { get; private set; }
+
+        /// <summary>
+        /// Gets the disabled storage account types.
+        /// </summary>
+        public StorageAccountTypes Supported { get; private set; }
```

Comment:
Let's not complicate the public UI for people with this - they shouldn't have to tell us what the account type is. I don't want our inability to determine a storage account type to leak through, since that will be temporary - we need to get a way to do this from storage (which we should/will be able to do eventually). We should push instead for that.